### PR TITLE
feat: DIA-1267: Allow support for Azure OpenAI connections

### PR DESCRIPTION
**Pull request by hakan458**
### PR fulfills these requirements
- [ ] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [ ] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [ ] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: label_studio/core/settings/base.py [hunk 1]
```diff
@@ -760,3 +760,5 @@ def collect_versions_dummy(**kwargs):
     'ssl_keyfile': get_env('REDIS_SSL_KEYFILE', None),
     'ssl_certfile': get_env('REDIS_SSL_CERTFILE', None),
 }
+
+OPENAI_API_VERSION = get_env('OPENAI_API_VERSION', '2024-06-01')
```
Comment (matt-bernstein) [inline label_studio/core/settings/base.py:764]: this is specific to azure right? Can you rename to imply that?

**Review by matt-bernstein — APPROVED** [review]

File: label_studio/ml_model_providers/migrations/0002_auto_20240718_1758.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        ),
+        migrations.AddField(
+            model_name='modelproviderconnection',
+            name='endpoint',
+            field=models.CharField(blank=True, help_text='Azure OpenAI endpoint', max_length=512, null=True),
+        ),
+        migrations.AlterField(
+            model_name='modelproviderconnection',
+            name='provider',
+            field=models.CharField(choices=[('OpenAI', 'OpenAI'), ('Azure', 'Azure')], default='OpenAI', max_length=255),
```
Comment (niklub) [inline label_studio/ml_model_providers/migrations/0002_auto_20240718_1758.py]: Would changing that field require DB remigration?

Comment (hakan458) [reply label_studio/ml_model_providers/migrations/0002_auto_20240718_1758.py]: Yes it seems every new choice we add will cause this small migration

File: label_studio/core/settings/base.py (hunk 1 above)
Comment (hakan458) [reply label_studio/core/settings/base.py:764]: We are only using it for AzureOpenAI client at the moment but its not Azure specific. Could be used elsewhere as well

File: label_studio/ml_model_providers/models.py [hunk 3]
```diff
@@ -6,19 +6,27 @@
 from django.utils.translation import gettext_lazy as _
 
 
-class ModelProviderConnection(models.Model):
-    class ModelProviders(models.TextChoices):
-        OPENAI = 'OpenAI', _('OpenAI')
+class ModelProviders(models.TextChoices):
+    OPENAI = 'OpenAI', _('OpenAI')
+    AZURE = 'Azure', _('Azure')
```
Comment (pakelley) [inline label_studio/ml_model_providers/models.py]: can we name this AZURE_OPENAI? There are other offerings from Azure, and it'd be nice not to have to go back and change this in various places if/when we add support for other azure services

**Review by pakelley — APPROVED** [review]
nothing blocking, lgtm

File: label_studio/ml_model_providers/models.py (hunk 3 above)
Comment (hakan458) [reply label_studio/ml_model_providers/models.py]: Yep makes sense

**Pull request merged**